### PR TITLE
DEPR: Deprecate Series/Dataframe.to_dense/to_sparse

### LINKED ISSUE #26557: DEPR: deprecate Series/DataFrame.to_dense/to_sparse
We should properly deprecate `to_dense` and `to_sparse` on Series and DataFrame.

`Series.to_dense` already is disfunctional (https://github.com/pandas-dev/pandas/issues/25680), and `DataFrame.to_dense` can be replaced with `DataFrame.sparse.to_dense()`.

And the `to_sparse` methods return a SparseSeries / SparseDataFrame, which are deprecated (and in that sense already show some deprecation warning, but this should be done properly with their own deprecation warning and doc update). 
Those could in principle be replaced with a new `DataFrame/Series.sparse.to_sparse` method? (although you can already do `astype('Sparse')`)

### PR DESCRIPTION
**Pull request by VikramjeetD**
- [x] closes  #26557 
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] add whatsnew entry
- [x] add tests

Marks the methods mentioned in #26557 as deprecated, with proper messages.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -1939,6 +1941,11 @@ def to_sparse(self, fill_value=None, kind='block'):
         >>> type(sdf)  # doctest: +SKIP
         <class 'pandas.core.sparse.frame.SparseDataFrame'>
         """
+        warning_message = """\
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py]:
```suggestion
        msg = "to_sparse is deprecated and will be removed in a future version"
```

File: pandas/core/series.py [hunk 2]
```diff
@@ -1573,6 +1573,7 @@ def to_frame(self, name=None):
 
     def to_sparse(self, kind='block', fill_value=None):
         """
+        ..deprecated:: 0.25.0
```
Comment (jorisvandenbossche) [inline pandas/core/series.py]: Here the same about putting it below the first line

File: pandas/core/sparse/frame.py [hunk 3]
```diff
@@ -277,6 +277,21 @@ def _unpickle_sparse_frame_compat(self, state):
 
     @Appender(SparseFrameAccessor.to_dense.__doc__)
     def to_dense(self):
+        """
+        ..deprecated:: 0.25.0
+        Use Dataframe.sparse.to_dense() instead
```
Comment (jorisvandenbossche) [inline pandas/core/sparse/frame.py]: This line needs to be indented with a few spaces (doesn't matter how many)

File: pandas/core/sparse/frame.py [hunk 4]
```diff
@@ -277,6 +277,21 @@ def _unpickle_sparse_frame_compat(self, state):
 
     @Appender(SparseFrameAccessor.to_dense.__doc__)
     def to_dense(self):
+        """
+        ..deprecated:: 0.25.0
```
Comment (jorisvandenbossche) [inline pandas/core/sparse/frame.py]:
```suggestion
        .. deprecated:: 0.25.0
```

File: pandas/core/sparse/frame.py [hunk 5]
```diff
@@ -277,6 +277,21 @@ def _unpickle_sparse_frame_compat(self, state):
...
+        Use Dataframe.sparse.to_dense() instead
+        """
+
+        warning_message = """\
+            to_dense is deprecated and will be removed in a future version
```
Comment (jorisvandenbossche) [inline pandas/core/sparse/frame.py]:
```suggestion
            SparseDataFrame.to_dense is deprecated and will be removed in a future version.
```

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR!

Some additional comments:

- There is an additional `to_dense` method in generic.py that should also be deprecated.
- You will need to add some tests that those warnings are raised (as an example see the test added here: https://github.com/pandas-dev/pandas/pull/26421/files#diff-f446e63e2ec57a5d055add5511588c3fR1234)

File: pandas/core/frame.py (hunk 1 above)
Comment (intEll1gent) [reply pandas/core/frame.py]: This change made `git diff upstream/master -u -- "*.py" | flake8 --diff` fail, modified it in a different way to do the job.

**Comment by intEll1gent** [comment]
@jorisvandenbossche Thanks for the review. I committed the stylistic changes that you suggested. 

Working on writing the tests. Could you please guide me on where the tests for Series and Dataframe (not sparse) need to be written? Should I create a new file or write it in an existing file.

File: pandas/core/frame.py [hunk 6]
```diff
@@ -1939,6 +1940,9 @@ def to_sparse(self, fill_value=None, kind='block'):
         >>> type(sdf)  # doctest: +SKIP
         <class 'pandas.core.sparse.frame.SparseDataFrame'>
         """
+        warnings.warn("DataFrame.to_sparse is deprecated and will be removed "
+                      "in a future version", FutureWarning, stacklevel=2)
+
         from pandas.core.sparse.api import SparseDataFrame
         return SparseDataFrame(self._series, index=self.index,
```
Comment (jreback) [inline pandas/core/frame.py]: wont' this *also* trigger the SDF warnings? (should this just be changed to create a DF here?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 6 above)
Comment (intEll1gent) [reply pandas/core/frame.py]: Do you mean the SDF deprecation warning? If thats the case, this was requested in the issue that this PR will address ( #26557 ).
Also, this might be helpful if someone has already hit the SDF warning through a different route, as a gentle reminder. Either way we could decide whats best and update accordingly.

Comment (jorisvandenbossche) [reply pandas/core/frame.py]: Yes, it already triggers the SparseDataFrame warning. But, it is still good to explicitly deprecate this method as well (in the docs, and by giving a clearer warning message, although the other one will also still be present).

I don't think we should change the behaviour, since we are deprecating it (it would also be a backwards incompatible change, as DataFrame with sparse and SparseDataFrame are not fully interchangeable).

**Comment by intEll1gent** [comment]
>   ...
> * There is an additional `to_dense` method in generic.py that should also be deprecated.
> ...

@jorisvandenbossche 
- Deprecating to_dense in `generic.py` breaks some tests for other classes. (I still haven't pushed that part). I could modify those tests but can't figure out how to ignore the FutureWarning. `@pytest.mark.filterwarnings('ignore::FutureWarning')` doesn't seem to help. 
- The other deprecations that I did raise a lot of warnings in many of the tests (but they pass)

I'll change the code style issues in the next push.

**Comment by jorisvandenbossche** [comment]
@intEll1gent I would need to look myself to say for sure (but now away for the weekend),  but if other tests raise those warnings, then it might be because the methods are still used in certain places in pandas. In such a case, you will need to check those places and see if it can be replaced with something else (or if it was just used in the tests directly, then you can add a filterwarnings decorator).

**Comment by intEll1gent** [comment]
> @intEll1gent I would need to look myself to say for sure (but now away for the weekend), but if other tests raise those warnings, then it might be because the methods are still used in certain places in pandas. In such a case, you will need to check those places and see if it can be replaced with something else (or if it was just used in the tests directly, then you can add a filterwarnings decorator).

Turns out replacing `tm.assert_produces_warning` with `pytest.warns` lets them pass.

File: pandas/core/frame.py (hunk 6 above)
Comment (intEll1gent) [reply pandas/core/frame.py]: So I think we can agree to keep both the warnings and resolve this conversation?

Comment (jreback) [reply pandas/core/frame.py]: actually i would suppress the SDF
warning here as the user facing to_sparse is already good enough

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 7]
```diff
@@ -219,6 +224,7 @@ def test_agg_multiple_functions_too_many_lambdas(df):
         grouped.agg(funcs)
 
 
+@pytest.mark.filterwarnings("ignore:NDFrame:FutureWarning")
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_aggregate.py]: did you add all of these? we certainly don't want to do this

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/aggregate/test_aggregate.py (hunk 7 above)
Comment (intEll1gent) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: I deprecated NDFrame.to_dense in generic.py, which caused warnings all throughout testing. Is there an alternative?

Comment (intEll1gent) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: @jreback Would it be fine if I filter warnings for only that method instead of all FutureWarnings, as such: `@pytest.mark.filterwarnings("ignore:NDFrame.to_dense:FutureWarning")`

Comment (jorisvandenbossche) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: Yes, the problem is that the current filter is too general, and it should filter only the warnings you are adding in this PR.

Comment (intEll1gent) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: @jorisvandenbossche gotcha, will update accordingly soon.

File: pandas/tests/arrays/test_integer.py [hunk 8]
```diff
@@ -659,6 +659,7 @@ def test_cross_type_arithmetic():
     tm.assert_series_equal(result, expected)
 
 
+@pytest.mark.filterwarnings("ignore:NDFrame.to_dense:FutureWarning")
```
Comment (jreback) [inline pandas/tests/arrays/test_integer.py]: I don't want to ignore these; instead if we are actually warning, then we need to adjust the internals to not do this (or *maybe* just catch the warning); these are essentially user facing tests

Comment (jreback) [reply pandas/tests/arrays/test_integer.py]: this comments applies to virtually all of the remaining filter warnings)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/arrays/test_integer.py (hunk 8 above)
Comment (intEll1gent) [reply pandas/tests/arrays/test_integer.py]: Not quite sure what you mean by user facing tests. This only ignores the warning during CI tests for pandas and not anything built using pandas. 

Also, I have written separate tests which specifically checks for and catches these warnings [here](https://github.com/pandas-dev/pandas/pull/26684/commits/50d0534f8c024686e2f939718c7e7e1cb19ab441#diff-227fb1652ca8fc0912cd32ed1d536297), so in case something goes wrong there, those tests would fail.

Comment (jreback) [reply pandas/tests/arrays/test_integer.py]: we do want to catch these warnings at all, rather to fix where they are coming from.

*only* a routine that is explicity calling ``.to_dense()`` should w catch, none of these tests do that. We are calling ``.to_dense()`` somewhere in the implementation; that needs to be fixed.

Comment (jorisvandenbossche) [reply pandas/tests/arrays/test_integer.py]: Those integer array groupby tests should indeed not generate a warning. @intEll1gent can you show the warning traceback that you get for the code of this test?

One tip can be to elevate the warning to an error to debug the location where it is coming from.

Comment (intEll1gent) [reply pandas/tests/arrays/test_integer.py]: The warning comes from usage of `to_dense` in `pandas/core/groupby/ops.py` lines 633, 635, 882. In fact, this seems to be the case for many of the other `NDFrame.to_dense` deprecation warnings. Removing `.to_dense` should do the job since the method simply returns whats passed in.

Comment (jorisvandenbossche) [reply pandas/tests/arrays/test_integer.py]: You might need to be careful about simply removing `to_dense`, because if you have sparse data in the DataFrame, then it actually does something.

Comment (intEll1gent) [reply pandas/tests/arrays/test_integer.py]: The methods I was thinking of changing seem to be operating on series. This is one of those three:
`class SeriesSplitter(DataSplitter):`
` def _chop(self, sdata, slice_obj):`
`return sdata._get_values(slice_obj).to_dense()`

I was thinking about replacing it with 
`class SeriesSplitter(DataSplitter):`
       `def _chop(self, sdata, slice_obj):`
            `return sdata._get_values(slice_obj)`

Local tests all pass with these changes.

Comment (intEll1gent) [reply pandas/tests/arrays/test_integer.py]: @jorisvandenbossche Are the changes I suggested in my previous comment fine or would it do something we don't want it to?

Comment (jorisvandenbossche) [reply pandas/tests/arrays/test_integer.py]: That's what I understood you proposed. If the tests pass, that is probably fine, so I would certainly push it (I was just wondering if one does a groupby with sparse columns, if the to_dense there would be needed, but then probably not)

File: pandas/core/frame.py [hunk 9]
```diff
@@ -2282,7 +2287,7 @@ def info(self, verbose=None, buf=None, max_cols=None, memory_usage=None,
         text_col     5 non-null object
         float_col    5 non-null float64
         dtypes: float64(1), int64(1), object(1)
-        memory usage: 200.0+ bytes
+        memory usage: 248.0+ bytes
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py]: You are changing this in several places. Is this related to this PR? 

(it seems that your changes are correct, but then we would prefer to have those changes in a separate PR if they are not related to the deprecation of to_sparse/to_dense)

File: pandas/core/generic.py [hunk 10]
```diff
@@ -1933,11 +1933,16 @@ def to_dense(self):
...
         Return dense representation of NDFrame (as opposed to sparse).
 
+        .. deprecated:: 0.25.0
+
         Returns
         -------
         %(klass)s
             Dense %(klass)s.
         """
+        warnings.warn("NDFrame.to_dense is deprecated "
```
[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply pandas/core/generic.py]: Can you apply this change?

File: pandas/core/sparse/series.py [hunk 11]
```diff
@@ -429,11 +431,15 @@ def _set_values(self, key, value):
     def to_dense(self):
         """
         Convert SparseSeries to a Series.
-
+        .. deprecated:: 0.25.0
```
Comment (jorisvandenbossche) [inline pandas/core/sparse/series.py]: Can you add a blank line above and below this one?

File: pandas/tests/io/json/test_pandas.py [hunk 12]
```diff
@@ -1018,13 +1018,19 @@ def test_sparse(self):
         df = pd.DataFrame(np.random.randn(10, 4))
         df.loc[:8] = np.nan
 
-        sdf = df.to_sparse()
+        # GH 26557: DEPR
+        with tm.assert_produces_warning(FutureWarning,
```
Comment (jorisvandenbossche) [inline pandas/tests/io/json/test_pandas.py]: I think here, you can also a filterwarning instead (since the test already has a filterwarning for the main Sparse deprecation as well)

File: pandas/tests/io/test_packers.py [hunk 13]
```diff
@@ -566,15 +566,24 @@ def test_sparse_series(self):
 
         s = tm.makeStringSeries()
         s[3:5] = np.nan
-        ss = s.to_sparse()
+        # GH 26557: DEPR
+        with tm.assert_produces_warning(FutureWarning,
```
Comment (jorisvandenbossche) [inline pandas/tests/io/test_packers.py]: The same for those tests, I think in general where there is a filterwarnings for Sparse, you can add one for the to_sparse methods as well

File: pandas/tests/io/json/test_pandas.py (hunk 12 above)
Comment (intEll1gent) [reply pandas/tests/io/json/test_pandas.py]: Will check on that, but we still have to come to an unanimous decision whether to keep the double deprecation warning for `Series/DataFrame.to_sparse`. None of this will be needed if we deicide to stick with only 1 warning for the main deprecation.

File: pandas/core/sparse/frame.py [hunk 14]
```diff
@@ -242,6 +242,11 @@ def _init_spmatrix(self, data, index, columns, dtype=None,
     def to_coo(self):
         return SparseFrameAccessor(self).to_coo()
 
+    def __repr__(self):
+        with warnings.catch_warnings():
```
Comment (jreback) [inline pandas/core/sparse/frame.py]: why are you adding this?

File: pandas/core/sparse/series.py [hunk 15]
```diff
@@ -214,10 +214,12 @@ def as_sparse_array(self, kind=None, fill_value=None, copy=False):
                            fill_value=fill_value, kind=kind, copy=copy)
 
     def __repr__(self):
-        series_rep = Series.__repr__(self)
-        rep = '{series}\n{index!r}'.format(series=series_rep,
-                                           index=self.sp_index)
-        return rep
+        with warnings.catch_warnings():
```
Comment (jreback) [inline pandas/core/sparse/series.py]: why are you adding this?

File: pandas/tests/arrays/sparse/test_arithmetics.py [hunk 16]
```diff
@@ -9,6 +9,7 @@
 
 
 @pytest.mark.filterwarnings("ignore:Sparse:FutureWarning")
+@pytest.mark.filterwarnings("ignore:Series.to_sparse:FutureWarning")
```
Comment (jreback) [inline pandas/tests/arrays/sparse/test_arithmetics.py]: where is this coming from?

File: pandas/tests/frame/test_deprecations.py [hunk 17]
```diff
@@ -0,0 +1,17 @@
+import numpy as np
+import pytest
+
```
Comment (jreback) [inline pandas/tests/frame/test_deprecations.py]: don't make a new file, just put it in an existing one

File: pandas/tests/generic/test_generic.py [hunk 18]
```diff
@@ -918,3 +918,16 @@ def test_axis_classmethods(self, box):
             assert obj._get_axis_name(v) == box._get_axis_name(v)
             assert obj._get_block_manager_axis(v) == \
                 box._get_block_manager_axis(v)
+
+    def test_deprecated_to_dense(self):
+        # Deprecated 0.25.0
```
Comment (jreback) [inline pandas/tests/generic/test_generic.py]: add the issue number

**Review by jreback — CHANGES_REQUESTED** [review]
again, I don't want blanket warning filters at all. These are hiding errors.

File: pandas/core/sparse/frame.py (hunk 14 above)
Comment (intEll1gent) [reply pandas/core/sparse/frame.py]: This wasn't added by me. It turned up in this PR after I did a pull from upstream towards the beginning of this PR. This is there in the upstream master branch.

File: pandas/core/sparse/series.py (hunk 15 above)
Comment (intEll1gent) [reply pandas/core/sparse/series.py]: Not added by me. Also present on upstream.

File: pandas/core/sparse/frame.py (hunk 14 above)
Comment (jorisvandenbossche) [reply pandas/core/sparse/frame.py]: @intEll1gent can you do a 

```
git fetch upstream master
git merge upstream/master
```

in your branch, and then push again? 
That should make sure that those changes are not visible here.

File: pandas/tests/arrays/sparse/test_arithmetics.py (hunk 16 above)
Comment (jorisvandenbossche) [reply pandas/tests/arrays/sparse/test_arithmetics.py]: From the Series.to_sparse calls in those tests.

**Comment by jorisvandenbossche** [comment]
> again, I don't want blanket warning filters at all. These are hiding errors.

For the old sparse tests (so not speaking about the groupby tests, there we indeed should certainly not hide the warnings), we decided earlier to not rewrite those tests but just ignore the warnings (until all functionality + tests can be removed). So you will see there are already existing filterwarnings for Sparse. For those cases (again, tests about the *old, deprecated* SparseDataFrame/Series, I think it is best to also ignore those new warnings).

**Comment by jorisvandenbossche** [comment]
> we still have to come to an unanimous decision whether to keep the double deprecation warning for Series/DataFrame.to_sparse. None of this will be needed if we deicide to stick with only 1 warning for the main deprecation.

@intEll1gent I think it is fine to have the double deprecation. How would you propose to only have a single warning? I personally think we certainly need a warning that mentions `to_sparse`, so that would mean suppressing the other warning from inside `to_sparse`?

**Comment by jreback** [comment]
> > again, I don't want blanket warning filters at all. These are hiding errors.
> 
> For the old sparse tests (so not speaking about the groupby tests, there we indeed should certainly not hide the warnings), we decided earlier to not rewrite those tests but just ignore the warnings (until all functionality + tests can be removed). So you will see there are already existing filterwarnings for Sparse. For those cases (again, tests about the _old, deprecated_ SparseDataFrame/Series, I think it is best to also ignore those new warnings).

right this case is ok, but we need to be really clear on this, e.g. on ``pandas/tests/sparse`` should have these types of filters

**Comment by jorisvandenbossche** [comment]
> Not added by me. Also present on upstream.

@intEll1gent for future reference: you are correct that it was not added by you, but in general, the diff that github shows on the PR needs to be correct. So if you noticed unrelated changes in the diff here from master, that means that something went wrong when you updated this branch to the latest master. And it is still up to you to fix that. See my earlier comment on how that could be done.

**Comment by jorisvandenbossche** [comment]
> right this case is ok, but we need to be really clear on this, e.g. on pandas/tests/sparse should have these types of filters

@jreback yes, that sounds correct, that we should in principle only do this general filterwarnings in the `pandas/tests/sparse`. However, it seems that with the original SparseDataFrame deprecation, such a general filterwarning was added in several other places as well (@TomAugspurger ? eg the `pandas/tests/arrays/sparse/test_arithmetic.py` do you remember why the general filterwarnings on Sparse was needed there?)

**Comment by jreback** [comment]
> > right this case is ok, but we need to be really clear on this, e.g. on pandas/tests/sparse should have these types of filters
> 
> @jreback yes, that sounds correct, that we should in principle only do this general filterwarnings in the `pandas/tests/sparse`. However, it seems that with the original SparseDataFrame deprecation, such a general filterwarning was added in several other places as well (@TomAugspurger ? eg the `pandas/tests/arrays/sparse/test_arithmetic.py` do you remember why the general filterwarnings on Sparse was needed there?)

right, let's comment these warning to be clear where they come from if they are not in pandas/tests/sparse (and actually maybe there too)

File: pandas/tests/frame/test_deprecations.py (hunk 17 above)
Comment (intEll1gent) [reply pandas/tests/frame/test_deprecations.py]: Any file? There doesn't seem to be a file in which this quite fits.

File: pandas/tests/io/json/test_pandas.py (hunk 12 above)
Comment (jreback) [reply pandas/tests/io/json/test_pandas.py]: > Will check on that, but we still have to come to an unanimous decision whether to keep the double deprecation warning for `Series/DataFrame.to_sparse`. None of this will be needed if we deicide to stick with only 1 warning for the main deprecation.

We should not keep the double warning at all.

**Comment by intEll1gent** [comment]
> > we still have to come to an unanimous decision whether to keep the double deprecation warning for Series/DataFrame.to_sparse. None of this will be needed if we deicide to stick with only 1 warning for the main deprecation.
> 
> @intEll1gent I think it is fine to have the double deprecation. How would you propose to only have a single warning? I personally think we certainly need a warning that mentions `to_sparse`, so that would mean suppressing the other warning from inside `to_sparse`?

Yes, I changed `to_sparse` to ignore the SDF/SS warnings and only show a single warning. The main warning shows in other cases.

File: pandas/core/sparse/frame.py [hunk 19]
```diff
@@ -282,6 +282,21 @@ def _unpickle_sparse_frame_compat(self, state):
...
+            Use Dataframe.sparse.to_dense() instead
+        """
+
+        warning_message = """\
+SparseDataFrame.to_dense is deprecated and will be removed in a future version
+
+            Use Dataframe.sparse.to_dense() instead
+
+            >>> df = pd.DataFrame({"A": pd.SparseArray([0, 1, 0])})
+            >>> df.sparse.to_dense()
```
Comment (jorisvandenbossche) [inline pandas/core/sparse/frame.py]: I was just thinking, maybe we don't need to deprecate this one, as it is on the SparseDataFrame class, which is already deprecated by itself?

File: pandas/tests/frame/test_deprecations.py (hunk 17 above)
Comment (jorisvandenbossche) [reply pandas/tests/frame/test_deprecations.py]: @intEll1gent I think a good place would be where the `to_sparse` functionality itself is tested (but don't know by heart where this is)

**Review by jorisvandenbossche — COMMENTED** [review]
Looking good!

File: pandas/core/sparse/frame.py (hunk 19 above)
Comment (intEll1gent) [reply pandas/core/sparse/frame.py]: Good point. Agreed. We probably don't need this.

File: pandas/tests/frame/test_deprecations.py (hunk 17 above)
Comment (intEll1gent) [reply pandas/tests/frame/test_deprecations.py]: There seems to be no test for `DataFrame.to_sparse()`? Github search for `test_to_sparse` in `pandas/tests` returns only 1 result, the one for `Series`.

File: pandas/tests/sparse/frame/test_frame.py [hunk 20]
```diff
@@ -350,6 +350,18 @@ def test_dense_to_sparse(self):
...
+        # GH 26557
+        # Deprecated 0.25.0
+
+        df = pd.DataFrame({"A": [1, np.nan, 3]})
+        sparse_df = pd.SparseDataFrame({"A": [1, np.nan, 3]})
+
+        with tm.assert_produces_warning(FutureWarning,
+                                        check_stacklevel=False):
+            result = df.to_sparse()
+        tm.assert_frame_equal(result, sparse_df)
```
Comment (jorisvandenbossche) [inline pandas/tests/sparse/frame/test_frame.py]: you put this in a class where those warnings you want to check for are generally ignored. Does that work? (just wondering)

Maybe we can also put it at the end of the file as a function instead of method, to avoid possible concerns about that.

Comment (jorisvandenbossche) [reply pandas/tests/sparse/frame/test_frame.py]: But since the tests are passing, this is probably fine ;)

**Review by jorisvandenbossche — APPROVED** [review]
Thanks, looks good!

@jreback you're fine with the current filterwarnings? They are only added there were we already had a filterwarning for other sparse warnings as well

cc @TomAugspurger

File: pandas/tests/sparse/frame/test_frame.py (hunk 20 above)
Comment (intEll1gent) [reply pandas/tests/sparse/frame/test_frame.py]: This actually works fine, `tm.assert_produces_warning` doesn't seem to care about filterwarnings. Remove the warning from `to_sparse` and this test fails. 

I wanted to keep both (`test_dense_to_sparse` and `test_deprecated_dense_to_sparse`) close for easy reference, thats all. Open to shifting it to the end though.

Comment (jorisvandenbossche) [reply pandas/tests/sparse/frame/test_frame.py]: Thanks for checking, then that sounds good!

**Comment by jreback** [comment]
i will review at some point again

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @intEll1gent !
